grails-app/domain/org/pih/warehouse/core/ProductPrice.groovy

```diff
@@ -0,0 +1,65 @@
+/**
+* Copyright (c) 2012 Partners In Health.  All rights reserved.
+* The use and distribution terms for this software are covered by the
+* Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+* which can be found in the file epl-v10.html at the root of this distribution.
+* By using this software in any fashion, you are agreeing to be bound by
+* the terms of this license.
+* You must not remove this notice, or any other, from this software.
+**/
+package org.pih.warehouse.core
+
+import org.pih.warehouse.auth.AuthService
+import org.pih.warehouse.product.ProductPackage
+
+class ProductPrice implements Serializable {
+
+    def beforeInsert = {
+        def currentUser = AuthService.currentUser.get()
+        if (currentUser) {
+            createdBy = currentUser
+            updatedBy = currentUser
+        }
+    }
+    def beforeUpdate = {
+        def currentUser = AuthService.currentUser.get()
+        if (currentUser) {
+            updatedBy = currentUser
+        }
+    }
+
+    String id
+
+    PriceTypeCode type = PriceTypeCode.DEFAULT_PRICE
+    BigDecimal price
+    UnitOfMeasure currency
+
+    Date startDate
+    Date endDate
+
+    // Audit fields
+    Date dateCreated
+    Date lastUpdated
+    User createdBy
+    User updatedBy
+
+    static belongsTo = [productPackage: ProductPackage]
```

Comment:
Do we need this? Or is it ok to just have ProductPrice on ProductPackage?

Reply:
I think we're looking for product_price_id to be added to the product_package table. So just make sure that is what's happening. A belongsTo would be used if we had multiple product prices associated with the product package.

Reply:
IMHO it is ok. I left this (and added productSupplier). Thanks to this we have cascade delete and save, which I think is expected in this case.

Reply:
Eventually the ProductPrice could have a hasMany association on Product, ProductPackage, ProductSupplier, so this belongsTo could grow to include more back references. In addition, we might need to change the associations to use a join table. If you're sure that keeping the belongsTo here doesn't paint us into a corner, then I'm ok with keeping it. I just don't want to add some minor convenience now that'll be painful to unravel later.